scipy/sparse/_sputils.py

```diff
@@ -241,14 +241,18 @@ def isintlike(x) -> bool:
     return True
 
 
-def isshape(x, nonneg=False, *, allow_1d=False) -> bool:
+def isshape(x, nonneg=False, *, allow_1d=False, allow_nd=False) -> bool:
     """Is x a valid tuple of dimensions?
 
     If nonneg, also checks that the dimensions are non-negative.
     If allow_1d, shapes of length 1 or 2 are allowed.
+    If allow_nd, shapes of all lengths >= 1 are allowed.
     """
     ndim = len(x)
-    if ndim != 2 and not (allow_1d and ndim == 1):
+    if ndim < 3:
+        if ndim != 2 and not (allow_1d and ndim == 1):
+            return False
+    elif not allow_nd:
```

Comment:
With just the allow_nd parameter this code could be simplified to something like:
```python
    if not allow_nd and ndim != 2:
        return False
```